tests/unit/LocalizeTests.ts

```diff
@@ -62,4 +64,130 @@ describe('localize', () => {
             expect(Localize.formatList(input)).toBe(expectedOutputES);
         });
     });
+
+    describe('translate method - missing translation behavior', () => {
+        beforeEach(async () => {
+            await IntlStore.load(CONST.LOCALES.EN);
+        });
+
+        it('should return MISSING_TRANSLATION for missing key with expensify email in production', async () => {
+            // Mock production environment
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            const CONFIG = require('@src/CONFIG');
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+            const originalConfig = {...CONFIG.default};
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            CONFIG.default.IS_IN_PRODUCTION = true;
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            CONFIG.default.IS_IN_STAGING = false;
+
+            // Set up session with expensify email
+            await Onyx.merge(ONYXKEYS.SESSION, {email: 'user@expensify.com'});
+            await waitForBatchedUpdates();
+
+            const result = Localize.translate(CONST.LOCALES.EN, 'missing.translation.key' as TranslationPaths);
+
+            expect(result).toBe(CONST.MISSING_TRANSLATION);
+
+            // Restore original config
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            Object.assign(CONFIG.default, originalConfig);
```

Comment:
This can also be DRYed up by moving it to `afterEach()`

Reply:
updated with helpers to simplify tests